src/proc_cpuview.c

```diff
@@ -776,21 +778,50 @@ int cpuview_proc_stat(const char *cg, const char *cpuset,
 	buf_size -= l;
 	total_len += l;
 
-	/* Render visible CPUs */
+	/* Render visible CPUs 
+	Assume there are K CPUs: 0, 1, 2, ..., K-1.
+	Among them, there are M online CPUs with index: a1, a2, ... aN ... aM (M >= N)
+	N = max_cpus, M = number of online CPUs
+
+	There will be N rendered cpus, indexed from 0 to N-1, cpu times of the cpus are calculated from those formula:
+	- user_time[0] = stat_node->view[0].user + stat_node->view[1].user + ... + stat_node->view[a1].user
+	- user_time[1] = stat_node->view[a1+1].user + stat_node->view[a1+1].user + ... + stat_node->view[a2].user
+	...
+	- user_time[N-2] = stat_node->view[a(N-2)+1].user + stat_node->view[a(N-2)+2].user + ... 
+	                    + stat_node->view[a(N-1)].user
+	- user_time[N-1] = stat_node->view[a(N-1)+1].user + stat_node->view[a(N-1)+2].user + ...
+	                    + stat_node->view[aN] + ... + stat_node->view[K-1] (sum of all remaining CPUs)
+	
+	Similar formula applied for system and idle time
+	*/
+
+	uint64_t curcpu_view_user_sum = 0, curcpu_view_system_sum = 0, curcpu_view_idle_sum = 0;
 	for (curcpu = 0, i = -1; curcpu < nprocs; curcpu++) {
-		if (!stat_node->usage[curcpu].online)
-			continue;
+		curcpu_view_user_sum += stat_node->view[curcpu].user;
+		curcpu_view_system_sum += stat_node->view[curcpu].system;
+		curcpu_view_idle_sum += stat_node->view[curcpu].idle;
 
+		if (!stat_node->usage[curcpu].online && curcpu < nprocs - 1) {
```

Comment:
Suppose this scenario:
nprocs is 64 while online CPUs is [0,1,2,3]
Then due to `curcpu < nprocs - 1` condition, when curcpu become 63, it would run `i++` again 
and then output `cpu4` in `/proc/stat`.

Here is an example
```
$ cat /sys/fs/cgroup/cpuset.cpus
0,1,2,3
$ cat /proc/stat
cpu  1379392 0 1359524 55515932 0 0 0 0 0 0
cpu0 344848 0 339881 13876474 0 0 0 0 0 0
cpu1 344848 0 339881 13880556 0 0 0 0 0 0
cpu2 344848 0 339881 13879745 0 0 0 0 0 0
cpu3 344848 0 339881 13879157 0 0 0 0 0 0
cpu4 0 0 0 0 0 0 0 0 0 0
```

Is it expected? @phanhuyn @brauner

Reply:
https://github.com/lxc/lxcfs/pull/544 resolve it.

Reply:
Thanks @likakuli